src/ui/screens/DemocracyScreen.tsx

```diff
@@ -99,6 +122,48 @@ export function DemocracyScreen() {
   );
 }
 
+type ProposalTeaserProps = {
+  proposal: Proposal;
+  children?: React.ReactNode;
+  navigation: NavigationProp<DashboardStackParamList>;
+};
+
+function DemocracyProposalTeaser({proposal, children, navigation}: ProposalTeaserProps) {
+  if (proposal.__typename === 'SubstrateChainDemocracyReferendum') {
```

Comment:
NIT: How about if we create a utility function like `function getTypeFrom(proposal: Proposal) =>  'proposal' | 'referendum';` ?

Reply:
In that case we loose the TS union discrimination. Is there a way make TS still infer the types using a util function.

Reply:
True, that was the point of using getting a `union` type from the api so we don't have to create unions in the client and use `__typename`. But for now we can live with that